lib/linalg/Matrix.h

```diff
@@ -1260,6 +1279,13 @@ class Matrix
      * If d_owns_data is false, then the object may not reallocate d_mat.
      */
     bool d_owns_data;
+
+    /**
+     * @brief If the absolute value of a double variable is below this
+     * tolerance, it is assumed to be zero.
+     * It is used to decide if we can safely divide by a value.
```

Comment:
On second thought, since this number is only used in the functions `orthogonalize` and `orthogonalize_last`, maybe it is better to make this an input parameter with default value, e.g.
`orthogonalize(double zero_tol=1.0e-15)`

Reply:
Done.